qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/RolesAndPrivilegesSteps.java

```diff
@@ -0,0 +1,162 @@
+    public void launchManagePrivilegesPage(){
+    	manageprivilegesPage = configuremetadatapage.goToManagePrivilegesPage();
+    }
+    
+    @When ("User clicks on Advanced Administration link from the System Administration Page")
+    public void launchAdvancedAdministrationPage(){
+    	administrationPage = systemAdministrationPage.goToAdvancedAdministration();
+    }
+    
+    @And ("User clicks the Add New Privilege button")
+    public void launchAddNewPrivilegePage(){
+    	addNewPrivilegePage = manageprivilegesPage.goToAddNewPrivilegePage();
+    }
+    
+    @And ("User fills the new privilege form")
+    public void addNewPrivilegeForm(){
+    	addNewPrivilegePage.enterPrivilegeName(PRIVILEGE_NAME);
+    	addNewPrivilegePage.enterPrivilegeDescription(PRIVILEGE_DESCRIPTION);
+    }
+    
+    @And ("User clicks the save button")
+    public void savePrivilege(){
+    	addNewPrivilegePage.clickSaveButton();
+    }
+    
+    @And ("User saves role")
+    public void saveRole(){
+    	addNewRolePage.saveRole();
+    }
+    
+    @And ("User fills the new role form")
+    public void addNewRoleForm(){
+    	addNewRolePage.addRoleName(ROLE_NAME);
+    	addNewRolePage.addDescription(ROLE_DESCRIPTION);
+    	addNewRolePage.selectPrivileges();
+    }
+    
+    @And ("User search for the created privilege")
+    public void searchPrivilege(){
+    	manageprivilegesPage.searchForPrivilege(PRIVILEGE_NAME);
+    }
+    
+    @And ("User edits privilege")
+    public void editPrivilegeForm(){
+    	manageprivilegesPage.goToEditPrivilegePage();
+    	addNewPrivilegePage.enterPrivilegeDescription("just testing the editing of a privilege");
+    	addNewPrivilegePage.clickSaveButton();
+    }
+    
+    @And ("User edits the role")
+    public void editRoleForm(){
+        administrationManageRolesPage.goToEditRolePage(ROLE_NAME);
+	addNewRolePage.addDescription("Developers of the OpenMRS...edited for the e2e automation test");
+	addNewRolePage.selectPrivileges();
+	addNewRolePage.saveRole();
+	}
+    
+    @And ("User clicks delete privilege")
+    public void deletePrivilege(){
+    	manageprivilegesPage.deletePrivilege();
+    }
+    
+    @And ("User clicks the Add New Role button on the manage roles page")
+    public void launchAddNewRolePage(){
+    	addNewRolePage = administrationManageRolesPage.goToAddNewRolePage();
+    }
+    
+    @And ("User clicks on manage roles link on the advanced administration page")
+    public void launchManageRolesPage(){
+    	administrationManageRolesPage = administrationPage.goToManageRolesPage();
+    }
+    
+    @Then ("System confirms delete")
+    public void confirmPrivilegeDeletion(){
+    	manageprivilegesPage.confirmPrivilegeDelete();
+    }
+    
+    @Then ("User deletes role")
+    public void deleteRole(){
+    	administrationManageRolesPage.deleteSelectedRoles();
+    }
+    
+    @Then ("Privilege is saved successfully")
+    public void privilegeSuccessfullySaved(){
```

Comment:
```suggestion
    public void privilegeSuccessfullySaved() {
```

Reply:
what do you think of `privilegeSuccessfullySaved()` vs `systemAddsPrivilege()` ?